app/controllers/declaration/FiscalInformationController.scala

```diff
@@ -35,32 +38,54 @@ class FiscalInformationController @Inject()(
   authenticate: AuthAction,
   journeyType: JourneyAction,
   customsCacheService: CustomsCacheService,
+  exportsCacheService: ExportsCacheService,
   mcc: MessagesControllerComponents
 )(implicit appConfig: AppConfig, ec: ExecutionContext)
-    extends FrontendController(mcc) with I18nSupport {
+    extends {
+  val cacheService = exportsCacheService
+} with FrontendController(mcc) with I18nSupport with ModelCacheable with SessionIdAware {
 
-  def displayPage(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+  def displayPage(itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
     customsCacheService.fetchAndGetEntry[FiscalInformation](goodsItemCacheId, formId).map {
-      case Some(data) => Ok(fiscal_information(form.fill(data)))
-      case _          => Ok(fiscal_information(form))
+      case Some(data) => Ok(fiscal_information(itemId, form.fill(data)))
+      case _          => Ok(fiscal_information(itemId, form))
     }
   }
 
-  def saveFiscalInformation(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    form()
-      .bindFromRequest()
-      .fold(
-        (formWithErrors: Form[FiscalInformation]) => Future.successful(BadRequest(fiscal_information(formWithErrors))),
-        validFiscalInformation =>
-          customsCacheService
-            .cache[FiscalInformation](goodsItemCacheId, formId, validFiscalInformation)
-            .map(_ => specifyNextPage(validFiscalInformation))
-      )
+  def saveFiscalInformation(itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async {
+    implicit request =>
+      form()
+        .bindFromRequest()
+        .fold(
+          (formWithErrors: Form[FiscalInformation]) =>
+            Future.successful(BadRequest(fiscal_information(itemId, formWithErrors))),
+          validFiscalInformation => updateCacheModels(itemId, validFiscalInformation)
+        )
   }
 
-  private def specifyNextPage(answer: FiscalInformation): Result =
+  private def updateCacheModels(itemId: String, validFiscalInformation: FiscalInformation)(
```

Comment:
Similar to comments above, I think it would be more intuitive if this just returns `Future[Unit]` rather than a `Future[Result]` given it is a method  called `updateCacheModels`

Minor though

Reply:
Swings and roundabouts... if we take the redirect out we end up with extra lines in the code. Might be better to rename the function but then you'd complain it's doing too much ;)

Reply:
I'll rename the function so it reflects what it is doing.